sycl/test-e2e/format.py

```diff
@@ -87,19 +76,60 @@ def parseTestScript(self, test):
 
         return script
 
-    def getMatchedFromList(self, features, alist):
+    def getMissingRequiredFeaturesFromList(
+        self, features, requires, build_only_mode=False
+    ):
         try:
             return [
-                item for item in alist if BooleanExpression.evaluate(item, features)
+                item
+                for item in requires
+                if not E2EExpr.evaluate(item, features, build_only_mode)
+            ]
+        except ValueError as e:
+            raise ValueError("Error in REQUIRES list:\n%s" % str(e))
+
+    def getMatchedFromList(self, features, alist, build_only_mode=False):
+        try:
+            return [
+                item
+                for item in alist
+                if E2EExpr.evaluate(item, features, build_only_mode, False)
             ]
         except ValueError as e:
             raise ValueError("Error in UNSUPPORTED list:\n%s" % str(e))
 
+    def select_triples_for_test(self, test):
+        supported_triples = set()
+        for t in test.config.sycl_triples:
+            features = test.config.available_features.union({t})
+            if self.getMissingRequiredFeaturesFromList(features, test.requires, build_only_mode=True):
+                continue
+            if self.getMatchedFromList(features, test.unsupported, build_only_mode=True):
+                continue
+            supported_triples.add(t)
+
+        if len(supported_triples) <= 1:
+            return supported_triples
+        # Treat XFAIL as UNSUPPORTED if the test is to be compiled for multiple
+        # triples.
+        if "*" in test.xfails:
```

Comment:
```suggestion
            return supported_triples
            
        # Treat XFAIL as UNSUPPORTED if the test is to be compiled for multiple
        # triples.
        
        if "*" in test.xfails:
```
because the comment refers to the rest of the function and not to the line 115 exclusively.